base_layer/wallet/src/key_manager_service/handle.rs

```diff
@@ -73,6 +73,22 @@ where TBackend: KeyManagerBackend + 'static
         (*self.key_manager_inner).write().await.apply_encryption(cipher)
     }
 
+    async fn create_key_pair<T: Clone + Into<String> + Send>(
+        &self,
+        branch: T,
+    ) -> Result<(PrivateKey, PublicKey), KeyManagerServiceError> {
+        self.key_manager_inner
+            .write()
+            .await
+            .add_key_manager_branch(branch.clone().into())?;
```

Comment:
You won't need to clone this, into returns an owned String
```suggestion
            .add_key_manager_branch(branch.into())?;
```

Reply:
Scratch that, I didnt see that you use the branch name below :)

Reply:
We need to clone the `branch` value, as we use it in line `85`. Better to clone it as a `String` as you suggested !